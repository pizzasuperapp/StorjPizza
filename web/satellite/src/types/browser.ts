// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all properties and methods present and available in the file/browser objects in Browser.
 */
export interface BrowserFile extends File {
  Key: string;
  LastModified: Date;
  Size: number;
  type: string;
}
