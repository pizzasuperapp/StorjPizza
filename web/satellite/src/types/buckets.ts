// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all bucket-related functionality.
 */
export interface BucketsApi {
    /**
     * Fetch buckets
     *
     * @returns BucketPage
     * @throws Error
     */
    get(projectId: string, before: Date, cursor: BucketCursor): Promise<BucketPage>;

    /**
     * Fetch all bucket names
     *
     * @returns string[]
     * @throws Error
     */
    getAllBucketNames(projectId: string): Promise<string[]>;
}

/**
 * Bucket class holds info for Bucket entity.
 */
export class Bucket {
    public constructor(
        public name: string = '',
        public storage: number = 0,
        public egress: number = 0,
        public objectCount: number = 0,
        public segmentCount: number = 0,
        public since: Date = new Date(),
        public before: Date = new Date(),
    ) { }
}

/**
 * BucketPage class holds bucket total usages and flag whether more usages available.
 */
export class BucketPage {
    public constructor(
        public buckets: Bucket[] = [],
        public search: string = '',
        public limit: number = 0,
        public offset: number = 0,
        public pageCount: number = 0,
        public currentPage: number = 0,
        public totalCount: number = 0,
    ) { }
}

/**
 * BucketCursor class holds cursor for bucket name and limit.
 */
export class BucketCursor {
    public constructor(
        public search: string = '',
        public limit: number = 0,
        public page: number = 0,
    ) { }
}
