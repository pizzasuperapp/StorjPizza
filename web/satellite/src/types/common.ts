// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export enum SortDirection {
    ASCENDING = 1,
    DESCENDING,
}

export enum OnboardingOS {
    WINDOWS = "windows",
    MAC = "macos",
    LINUX = "linux",
}

export class PartneredSatellite {
    constructor(
        public name: string = '',
        public address: string = '',
    ) {}
}
