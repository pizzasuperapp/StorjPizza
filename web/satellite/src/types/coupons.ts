// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

export enum CouponType {
	NoCoupon = "noCoupon",
	FreeTierCoupon = "freeTierCoupon",
	SignupCoupon = "signupCoupon"
}
