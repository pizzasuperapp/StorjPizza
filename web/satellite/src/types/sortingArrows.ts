// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export enum SortingDirectionEnum {
    TOP = 1,
    BOTTOM,
}
