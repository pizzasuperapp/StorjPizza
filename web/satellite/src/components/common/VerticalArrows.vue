// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="container">
        <TopArrowIcon :class="{ active: isActive && isTop }" />
        <BottomArrowIcon :class="{ active: isActive && isBottom }" />
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import BottomArrowIcon from '@/../static/images/common/bottomArrow.svg';
import TopArrowIcon from '@/../static/images/common/topArrow.svg';

import { SortingDirectionEnum } from '@/types/sortingArrows';

// @vue/component
@Component({
    components: {
        TopArrowIcon,
        BottomArrowIcon,
    },
})
export default class VerticalArrows extends Vue {
    @Prop({default: false})
    private isActive: boolean;
    @Prop({default: SortingDirectionEnum.BOTTOM})
    private direction: SortingDirectionEnum;

    public get isTop(): boolean {
        return this.direction === SortingDirectionEnum.TOP;
    }

    public get isBottom(): boolean {
        return this.direction === SortingDirectionEnum.BOTTOM;
    }
}
</script>

<style scoped lang="scss">
    .container {
        display: flex;
        flex-direction: column;
        margin-left: 10px;
        justify-content: space-between;
        height: 17px;
    }

    .active {

        .arrow-svg-path {
            fill: #2683ff !important;
        }
    }
</style>
