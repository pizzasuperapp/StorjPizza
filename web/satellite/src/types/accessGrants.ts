// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import { SortDirection } from '@/types/common';

export type OnHeaderClickCallback = (sortBy: AccessGrantsOrderBy, sortDirection: SortDirection) => Promise<void>;

/**
 * AccessGrantsWorker provides access to the WASM module.
 */
export interface AccessGrantsWorkerFactory {
    // TODO: this should be converted to a proper interface.
    create(): Worker;
}

/**
 * Exposes all access grants-related functionality.
 */
export interface AccessGrantsApi {
    /**
     * Fetch accessGrants
     *
     * @returns AccessGrantsPage
     * @throws Error
     */
    get(projectId: string, cursor: AccessGrantCursor): Promise<AccessGrantsPage>;

    /**
     * Create new accessGrant
     *
     * @returns AccessGrant
     * @throws Error
     */
    create(projectId: string, name: string): Promise<AccessGrant>;

    /**
     * Delete existing access grant
     *
     * @returns null
     * @throws Error
     */
    delete(ids: string[]): Promise<void>;

    /**
     * Delete existing access grant by name and project id
     *
     * @returns null
     * @throws Error
     */
    deleteByNameAndProjectID(name: string, projectID: string): Promise<void>;

    /**
     * Get gateway credentials using access grant
     *
     * @returns EdgeCredentials
     * @throws Error
     */
    getGatewayCredentials(accessGrant: string, optionalURL?: string, isPublic?: boolean): Promise<EdgeCredentials>;
}

/**
 * Holds access grants sorting parameters.
 */
export enum AccessGrantsOrderBy {
    NAME = 1,
    CREATED_AT,
}

/**
 * AccessGrantCursor is a type, used to describe paged access grants list.
 */
export class AccessGrantCursor {
    public constructor(
        public search: string = '',
        public limit: number = 6,
        public page: number = 1,
        public order: AccessGrantsOrderBy = AccessGrantsOrderBy.NAME,
        public orderDirection: SortDirection = SortDirection.ASCENDING,
    ) {}
}

/**
 * AccessGrantsPage is a type, used to describe paged access grants list.
 */
export class AccessGrantsPage {
    public constructor(
        public accessGrants: AccessGrant[] = [],
        public search: string = '',
        public order: AccessGrantsOrderBy = AccessGrantsOrderBy.NAME,
        public orderDirection: SortDirection = SortDirection.ASCENDING,
        public limit: number = 6,
        public pageCount: number = 0,
        public currentPage: number = 1,
        public totalCount: number = 0,
    ) {}
}

/**
 * AccessGrant class holds info for Access Grant entity.
 */
export class AccessGrant {
    public isSelected: boolean;

    constructor(
        public id: string = '',
        public name: string = '',
        public createdAt: Date = new Date(),
        public secret: string = '',
    ) {
        this.isSelected = false;
    }

    /**
     * Returns created date as a local string.
     */
    public localDate(): string {
        return this.createdAt.toLocaleString('en-US', {timeZone: 'UTC', timeZoneName: 'short'});
    }
}

/**
 * DurationPermission class holds info for access grant's duration permission.
 */
export class DurationPermission {
    constructor(
        public notBefore: Date | null = null,
        public notAfter: Date | null = null,
    ) {}
}

/**
 * EdgeCredentials class holds info for edge credentials generated from access grant.
 */
export class EdgeCredentials {
    constructor(
        public id: string = '',
        public createdAt: Date = new Date(),
        public accessKeyId: string = '',
        public secretKey: string = '',
        public endpoint: string = '',
    ) {}
}
