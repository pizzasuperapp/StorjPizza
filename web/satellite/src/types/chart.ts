// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

import { DataStamp } from "@/types/projects";
import { Size } from "@/utils/bytesSize";

/**
 * ChartData class holds info for ChartData entity.
 */
export class ChartData {
    public labels: string[];
    public datasets: DataSets[] = [];

    public constructor(
        labels: string[],
        backgroundColor: string,
        borderColor: string,
        pointBorderColor: string,
        data: number[],
        secondaryBackgroundColor?: string,
        secondaryBorderColor?: string,
        secondaryPointBorderColor?: string,
        secondaryData?: number[],
    ) {
        this.labels = labels;
        this.datasets[0] = new DataSets(backgroundColor, borderColor, pointBorderColor, data)

        if (secondaryData && secondaryBackgroundColor && secondaryBorderColor && secondaryPointBorderColor) {
            this.datasets[1] = new DataSets(secondaryBackgroundColor, secondaryBorderColor, secondaryPointBorderColor, secondaryData);
        }
    }
}

/**
 * DataSets class holds info for chart's DataSets entity.
 */
class DataSets {
    public constructor(
        public backgroundColor: string,
        public borderColor: string,
        public pointBorderColor: string,
        public data: number[],
        public borderWidth: number = 4,
        public pointHoverBackgroundColor: string = 'white',
        public pointHoverBorderWidth: number = 5,
    ) {}
}

/**
 * TooltipParams holds tooltip's configuration
 */
export class TooltipParams {
    public constructor(
        public tooltipModel: TooltipModel,
        public chartId: string,
        public tooltipId: string,
        public markUp: string,
        public tooltipTop: number,
        public tooltipLeft: number,
    ) {}
}

/**
 * StylingConstants holds tooltip styling constants
 */
class StylingConstants {
    public static tooltipOpacity = '1';
    public static tooltipPosition = 'absolute';
}

/**
 * Styling holds tooltip's styling configuration
 */
class Styling {
    public constructor(
        public tooltipModel: TooltipModel,
        public element: HTMLElement,
        public topPosition: number,
        public leftPosition: number,
        public chartPosition: ClientRect,
    ) {}
}

/**
 * Color is a color definition.
 */
export type Color = string

/**
 * TooltipItem contains datapoint information.
 */
export interface TooltipItem {
    // Label for the tooltip
    label: string,

    // Value for the tooltip
    value: string,

    // X Value of the tooltip
    // (deprecated) use `value` or `label` instead
    xLabel: number | string,

    // Y value of the tooltip
    // (deprecated) use `value` or `label` instead
    yLabel: number | string,

    // Index of the dataset the item comes from
    datasetIndex: number,

    // Index of this data item in the dataset
    index: number,

    // X position of matching point
    x: number,

    // Y position of matching point
    y: number
}

/**
 * TooltipModel contains parameters that can be used to render the tooltip.
 */
export interface TooltipModel {
    // The items that we are rendering in the tooltip. See Tooltip Item Interface section
    dataPoints: TooltipItem[],

    // Positioning
    xPadding: number,
    yPadding: number,
    xAlign: string,
    yAlign: string,

    // X and Y properties are the top left of the tooltip
    x: number,
    y: number,
    width: number,
    height: number,
    // Where the tooltip points to
    caretX: number,
    caretY: number,

    // Body
    // The body lines that need to be rendered
    // Each object contains 3 parameters
    // before: string[] // lines of text before the line with the color square
    // lines: string[], // lines of text to render as the main item with color square
    // after: string[], // lines of text to render after the main lines
    body: {before: string[]; lines: string[], after: string[]}[],
    // lines of text that appear after the title but before the body
    beforeBody: string[],
    // line of text that appear after the body and before the footer
    afterBody: string[],
    bodyFontColor: Color,
    _bodyFontFamily: string,
    _bodyFontStyle: string,
    _bodyAlign: string,
    bodyFontSize: number,
    bodySpacing: number,

    // Title
    // lines of text that form the title
    title: string[],
    titleFontColor: Color,
    _titleFontFamily: string,
    _titleFontStyle: string,
    titleFontSize: number,
    _titleAlign: string,
    titleSpacing: number,
    titleMarginBottom: number,

    // Footer
    // lines of text that form the footer
    footer: string[],
    footerFontColor: Color,
    _footerFontFamily: string,
    _footerFontStyle: string,
    footerFontSize: number,
    _footerAlign: string,
    footerSpacing: number,
    footerMarginTop: number,

    // Appearance
    caretSize: number,
    caretPadding: number,
    cornerRadius: number,
    backgroundColor: Color,

    // colors to render for each item in body[]. This is the color of the squares in the tooltip
    labelColors: Color[],
    labelTextColors: Color[],

    // 0 opacity is a hidden tooltip
    opacity: number,
    legendColorBackground: Color,
    displayColors: boolean,
    borderColor: Color,
    borderWidth: number
}

/**
 * Tooltip provides custom tooltip rendering
 */
export class Tooltip {
    public static custom(params: TooltipParams): void {
        const chart = document.getElementById(params.chartId);

        if (!chart) {
            return;
        }

        const tooltip: HTMLElement = Tooltip.createTooltip(params.tooltipId);

        if (!params.tooltipModel.opacity) {
            Tooltip.remove(tooltip);

            return;
        }

        if (params.tooltipModel.body) {
            Tooltip.render(tooltip, params.markUp);
        }

        const position = chart.getBoundingClientRect();

        const tooltipStyling = new Styling(params.tooltipModel, tooltip, params.tooltipTop, params.tooltipLeft, position);
        Tooltip.elemStyling(tooltipStyling);
    }

    public static createTooltip(id: string): HTMLElement {
        let tooltipEl = document.getElementById(id);

        if (!tooltipEl) {
            tooltipEl = document.createElement('div');
            tooltipEl.id = id;
            document.body.appendChild(tooltipEl);
        }

        return tooltipEl;
    }

    public static remove(tooltipEl: HTMLElement): void {
        document.body.removeChild(tooltipEl);
    }

    private static render(tooltip: HTMLElement, markUp: string): void {
        tooltip.innerHTML = markUp;
    }

    private static elemStyling(elemStyling: Styling): void {
        elemStyling.element.style.opacity = StylingConstants.tooltipOpacity;
        elemStyling.element.style.position = StylingConstants.tooltipPosition;
        elemStyling.element.style.left = `${elemStyling.chartPosition.left + elemStyling.tooltipModel.caretX - elemStyling.leftPosition}px`;
        elemStyling.element.style.top = `${elemStyling.chartPosition.top + window.pageYOffset + elemStyling.tooltipModel.caretY - elemStyling.topPosition}px`;
    }
}

/**
 * Stores data for chart's tooltip
 */
export class ChartTooltipData {
    public date: string;
    public value: string;

    public constructor(stamp: DataStamp) {
        const size = new Size(stamp.value, 1)

        this.date = stamp.intervalStart.toLocaleDateString('en-US', { day: '2-digit', month: 'short' });
        this.value = `${size.formattedBytes} ${size.label}`;
    }
}

/**
 * RenderChart contains definition for renderChart and addPlugin, that can be used to cast
 * a derived chart type, with `(this as unknown as RenderChart).renderChart`
 */
export interface RenderChart {
    renderChart<A, B>(A, B): void
    addPlugin (plugin?: Record<string, (chart) => void>): void
}
