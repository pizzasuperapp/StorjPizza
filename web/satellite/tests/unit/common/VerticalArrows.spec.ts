// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import VerticalArrows from '@/components/common/VerticalArrows.vue';

import { SortingDirectionEnum } from '@/types/sortingArrows';
import { mount } from '@vue/test-utils';

describe('VerticalArrows.vue', () => {
    it('should render with bottom arrow highlighted', function () {
        const wrapper = mount(VerticalArrows, {
            propsData: {
                isActive: true,
                direction: SortingDirectionEnum.BOTTOM,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render without any highlighted arrows', function () {
        const wrapper = mount(VerticalArrows, {
            propsData: {
                isActive: false,
                direction: SortingDirectionEnum.BOTTOM,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render with top arrow highlighted', function () {
        const wrapper = mount(VerticalArrows, {
            propsData: {
                isActive: true,
                direction: SortingDirectionEnum.TOP,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
