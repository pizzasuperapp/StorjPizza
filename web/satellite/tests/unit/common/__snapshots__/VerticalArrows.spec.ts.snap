// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalArrows.vue should render with bottom arrow highlighted 1`] = `<div class="container"><svg class=""></svg> <svg class="active"></svg></div>`;

exports[`VerticalArrows.vue should render with top arrow highlighted 1`] = `<div class="container"><svg class="active"></svg> <svg class=""></svg></div>`;

exports[`VerticalArrows.vue should render without any highlighted arrows 1`] = `<div class="container"><svg class=""></svg> <svg class=""></svg></div>`;
